### PR TITLE
Docs, splits and simplifications

### PR DESCRIPTION
**Pull request by psychon**
This PR brings `#[deny(missing_docs)]` to xcbgen-rs. It also splits up one of the largest files in there into multiple sub-files (no idea if I chose good boundaries for these submodules). Finally, this also brings a minor simplification.

All of this has no effect on the generated code.

### CODE REVIEW

File: xcbgen-rs/src/defs.rs [hunk 1]
```diff
@@ -1827,111 +2026,223 @@ impl BuiltInType {
...
+    /// The name of the list that is used.
     pub field_name: String,
+
+    /// After resolving, a reference to the list that is used.
     pub resolved_field: OnceCell<ResolvedFieldRef>,
+
+    /// The expression that is applied to each element of the list.
+    ///
+    /// If no expression is provided, the elements are summed as-is.
+    // FIXME: Can the Option be replaced with `Expression::ListElementRef`?
```
Comment (eduardosm) [inline xcbgen-rs/src/defs.rs]: It seems it can be replaced, I'll open a PR for that.

File: xcbgen-rs/src/defs.rs [hunk 2]
```diff
@@ -1827,111 +2026,223 @@ impl BuiltInType {
...
     pub fields: Vec<FieldDoc>,
+
+    /// The possible X11 errors that can occur.
     pub errors: Vec<ErrorDoc>,
+
+    /// A reference to a related type.
     pub sees: Vec<SeeDoc>,
 }
 
+/// Documentation for a single type.
```
Comment (eduardosm) [inline xcbgen-rs/src/defs.rs]: Did you mean a single field?

File: xcbgen-rs/src/defs/expression.rs [hunk 3]
```diff
@@ -165,6 +165,5 @@ pub struct SumOfExpr {
     /// The expression that is applied to each element of the list.
     ///
     /// If no expression is provided, the elements are summed as-is.
-    // FIXME: Can the Option be replaced with `Expression::ListElementRef`?
-    pub operand: Option<Box<Expression>>,
+    pub operand: Box<Expression>,
```
Comment (eduardosm) [inline xcbgen-rs/src/defs/expression.rs:168]: Oh, you already did it

File: xcbgen-rs/src/defs.rs (hunk 1 above)
Comment (psychon) [reply xcbgen-rs/src/defs.rs]: Already done in a later commit in this PR.

File: xcbgen-rs/src/defs.rs (hunk 2 above)
Comment (psychon) [reply xcbgen-rs/src/defs.rs]: Uhm... yes, I guess so.

**Pull request merged**